tools/spy-detector/src/main.rs

```diff
@@ -0,0 +1,262 @@
+use dashmap::DashMap;
+use nng::options::protocol::pubsub::Subscribe;
+use nng::options::Options;
+use nng::{Protocol, Socket};
+use shared::clap;
+use shared::clap::Parser;
+use shared::event_msg;
+use shared::event_msg::event_msg::Event;
+use shared::net_msg;
+use shared::prost::Message;
+use shared::simple_logger;
+use std::sync::atomic;
+use std::sync::atomic::AtomicU32;
+use std::sync::Arc;
+use std::time::Duration;
+use std::time::Instant;
+
+const ADDRESS: &str = "tcp://127.0.0.1:8883";
+const STATS_INTERVAL: Duration = Duration::from_secs(10); // Duration(seconds) to print stats of peers
```

Comment:
probably also good to have this as an argument

Reply:
IIRC we discussed printing every 2 minutes at one point - do you have a reason for using 10s now?

Reply:
Done!!